src/apps/api/views/leaderboards.py

```diff
@@ -47,7 +47,7 @@ def add_submission_to_leaderboard(request, submission_pk):
     submission = get_object_or_404(Submission, pk=submission_pk)
 
     # Removing any existing submissions on leaderboard
-    submission.phase.submissions.filter(owner=request.user).exclude(leaderboard=None).update(leaderboard=None)
+    Submission.objects.filter(owner=request.user).update(leaderboard=None)
```

Comment:
this will remove all submissions from the leaderboard, we need to also filter on phase (see Brendon's PR: https://github.com/codalab/competitions-v2/pull/455/files)